src/sofa/pbrpc/rpc_server_impl.cc

```diff
@@ -182,11 +182,11 @@ void RpcServerImpl::Stop()
     _timer_worker->stop();
     _listener->close();
     StopStreams();
-    _work_thread_group->stop();
 
     _timer_worker.reset();
     _listener.reset();
     ClearStreams();
+    _work_thread_group->stop();
     _maintain_thread_group->stop();
 
```

Comment:
Be careful to change the order. You'd better run test/kill_test for a long time to check if new bug is brought in.